staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -491,14 +487,35 @@ func (c *Cacher) Watch(ctx context.Context, key string, opts storage.ListOptions
 		return nil, errors.NewServiceUnavailable(err.Error())
 	}
 
+	// Making triggerSupported be true requires the following conditions to be
```

Comment:
I think the usecase you're trying to address is basically "I want to watch individual SAs". In other words "watch per service-account".

For that you actually need an index on two fields (namespace, name), not two separate indexes on two fields.

The individual indexes are by definition independent, so to achieve (namespace, name) you effectively will be forced to to join the indexes, and we definitely don't want that.

So unless I misunderstood something, none of these changes aren't really needed.
All you need to do really is to define an index and trigger function to be sth like:
(metadata.namespace, metadata.name) and produce a value that is effectively some kind of concatenation of those two.

Reply:
Thank you very much, Wojciech. Following your comments, I rethought the whole logic and added an extra field selector which uses "metadata.namespace/metadata.name". Previously I was not familiar with the usage of field selector. Now I understand it.